File: tmp-fix-api-urls.py
from pathlib import Path
import re

path = Path('src/App.jsx')
text = path.read_text(encoding='utf-8')
pattern = re.compile(r"(['\"`])http://localhost:5000([^'\"`]*)\1")
new_text = pattern.sub(lambda m: f"`\${{apiBaseUrl}}{m.group(2)}`", text)
path.write_text(new_text, encoding='utf-8')
print('changed' if new_text != text else 'unchanged')

File: tmp-fix-api-urls.js
const fs = require('fs');
const path = require('path');
const filePath = path.join('src', 'App.jsx');
let text = fs.readFileSync(filePath, 'utf8');
const original = text;
text = text.replace(/(['"`])http:\/\/localhost:5000([^'"`]*)\1/g, (match, quote, tail) => {
  if (quote === '`') {
    return `\${apiBaseUrl}${tail}`;
  }
  return `\${apiBaseUrl}${tail}`;
});
if (text !== original) {
  fs.writeFileSync(filePath, text, 'utf8');
  console.log('Replaced localhost API URLs in src/App.jsx.');
} else {
  console.log('No replacements were made.');
}
